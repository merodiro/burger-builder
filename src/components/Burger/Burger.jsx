import React from 'react'

import classes from './Burger.css'
import Ingredient from './Ingredient/Ingredient'

export default ({ ingredients }) => {
  let transformed = Object.keys(ingredients)
    .map(key =>
      [...Array(ingredients[key])].map((_, i) => (
        <Ingredient type={key} key={key + i} />
      ))
    )
    .reduce((arr, el) => arr.concat(el), [])

  if (transformed.length === 0) {
    transformed = <p>Please start adding ingredients!</p>
  }
  return (
    <div className={classes.Burger}>
      <Ingredient type="bread-top" />
      {transformed}
      <Ingredient type="bread-bottom" />
    </div>
  )
}
