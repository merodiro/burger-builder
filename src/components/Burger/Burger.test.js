import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'

import Burger from './Burger'

describe('<Burger>', () => {
  it('renders the same output', () => {
    const ingredients = {
      salad: 0,
      bacon: 0,
      cheese: 0,
      meat: 0,
    }
    const component = shallow(<Burger ingredients={ingredients} />)

    expect(component).toMatchSnapshot()
  })

  it('renders with ingredients', () => {
    const ingredients = {
      salad: 3,
      bacon: 1,
      cheese: 2,
      meat: 1,
    }
    const component = shallow(<Burger ingredients={ingredients} />)

    expect(component).toMatchSnapshot()
  })
})
