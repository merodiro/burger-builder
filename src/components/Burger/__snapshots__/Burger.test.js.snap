// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Burger> renders the same output 1`] = `
<div
  className="Burger"
>
  <Ingredient
    type="bread-top"
  />
  <p>
    Please start adding ingredients!
  </p>
  <Ingredient
    type="bread-bottom"
  />
</div>
`;

exports[`<Burger> renders with ingredients 1`] = `
<div
  className="Burger"
>
  <Ingredient
    type="bread-top"
  />
  <Ingredient
    key="salad0"
    type="salad"
  />
  <Ingredient
    key="salad1"
    type="salad"
  />
  <Ingredient
    key="salad2"
    type="salad"
  />
  <Ingredient
    key="bacon0"
    type="bacon"
  />
  <Ingredient
    key="cheese0"
    type="cheese"
  />
  <Ingredient
    key="cheese1"
    type="cheese"
  />
  <Ingredient
    key="meat0"
    type="meat"
  />
  <Ingredient
    type="bread-bottom"
  />
</div>
`;
